### PR TITLE
Update to 1.12

### PR DESCRIPTION
**Pull request by gudenau**
I messed with the build file quite a bit, I do not know gradle that well so sorry if you have to fix anything.

Solves #40

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -89,107 +62,26 @@ minecraft {
...
+    // copy everything else except the mcmod.info
     from(sourceSets.main.resources.srcDirs) {
         exclude 'mcmod.info'
     }
 }
-
-task deobfJar(type: Jar) {
-    from sourceSets.main.output
-    classifier = 'deobf'
-}
```
Comment (KnightMiner) [inline build.gradle]: Why was all this deleted?

File: src/main/java/slimeknights/toolleveling/CommonProxy.java [hunk 2]
```diff
@@ -8,34 +8,45 @@
...
+@Mod.EventBusSubscriber(modid = MODID)
+public class CommonProxy{
+    static SoundEvent SOUND_LEVELUP;
 
-  public void playLevelupDing(EntityPlayer player) {
-    Sounds.PlaySoundForPlayer(player, SOUND_LEVELUP, 1f, 1f);
-  }
+    public void playLevelupDing(EntityPlayer player){
+        Sounds.PlaySoundForPlayer(player, SOUND_LEVELUP, 1f, 1f);
+    }
```
Comment (KnightMiner) [inline src/main/java/slimeknights/toolleveling/CommonProxy.java]: Please don't include formatting changes, they make it difficult to tell what actually changed

File: src/main/java/slimeknights/toolleveling/config/ConfigGui.java [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+import net.minecraftforge.fml.client.config.IConfigElement;
+import slimeknights.toolleveling.TinkerToolLeveling;
+
+import java.util.List;
+
+/**
+ * Created by gudenau on 8/7/2017.
+ * <p>
+ * TinkersToolLeveling
+ */
```
Comment (KnightMiner) [inline src/main/java/slimeknights/toolleveling/config/ConfigGui.java]: Typically we don't have created by comments, thats what the git history is for

File: src/main/java/slimeknights/toolleveling/config/ConfigSync.java [hunk 4]
```diff
@@ -1,23 +0,0 @@
-package slimeknights.toolleveling.config;
```
Comment (KnightMiner) [inline src/main/java/slimeknights/toolleveling/config/ConfigSync.java]: Any reason this was removed? I don't see ConfigSyncing not being required anymore

File: build.gradle (hunk 1 above)
Comment (gudenau) [inline build.gradle]: I remade the build.gradle, it was not working and I am not good with it. Just added stuff back until I got it to work decently for the most part.

File: src/main/java/slimeknights/toolleveling/CommonProxy.java (hunk 2 above)
Comment (gudenau) [inline src/main/java/slimeknights/toolleveling/CommonProxy.java]: That was my IDE.

File: src/main/java/slimeknights/toolleveling/config/ConfigGui.java (hunk 3 above)
Comment (gudenau) [inline src/main/java/slimeknights/toolleveling/config/ConfigGui.java]: That was my IDE.

File: src/main/java/slimeknights/toolleveling/config/ConfigSync.java (hunk 4 above)
Comment (gudenau) [inline src/main/java/slimeknights/toolleveling/config/ConfigSync.java]: I had issues with the config code, remade it. Worked for me, might need tweaking though.

File: src/main/java/slimeknights/toolleveling/CommonProxy.java (hunk 2 above)
Comment (KnightMiner) [inline src/main/java/slimeknights/toolleveling/CommonProxy.java]: You should be able to turn off autoformatting, or modify the formatter to match this project. Thats what I've done for Tinkers Construct

File: src/main/java/slimeknights/toolleveling/config/ConfigSync.java (hunk 4 above)
Comment (KnightMiner) [inline src/main/java/slimeknights/toolleveling/config/ConfigSync.java]: The point of this was server to client syncing if I understand it correctly, does that function in the new code?

Comment (gudenau) [inline src/main/java/slimeknights/toolleveling/config/ConfigSync.java]: Probably not, I will play with that tomorrow.

**Comment by octoshrimpy** [comment]
any news?

**Comment by gudenau** [comment]
Sever syncing is a little annoying, I'll have to take a crack at it with a fresh set of eyes.

**Comment by bonii-xx** [comment]
Regarding the syncing issues, it's most likely this: https://github.com/SlimeKnights/Mantle/issues/93

**Comment by kenijey** [comment]
hey ah, how is it going about this 1.12 port? just wondering will we get a 1.12 port of this mod

**Comment by gudenau** [comment]
Sorry, I've been busy.

**Comment by kenijey** [comment]
so we will see a port? :)

**Comment by bonii-xx** [comment]
The mantle update did solve the guava problems. I took your and @Nettifani's changes and updated it manually quickly, since it was faster that way. And less changes.

**Pull request closed without merging**